func3d.py

```diff
@@ -90,6 +90,85 @@ def pgradient(self, rvec):
         akderiv = 2*a/(1+a)**3 * r
         return akderiv
 
+    
+class ExpCuspFunction:
+    """
+    :math:`b(r) = \frac{c p(r/r_{cut})}{1+\gamma*p(r/r_{cut})}` 
+    where 
+    :math:`p(y) = y - y^2 + y^3/3`
+    """
+    def __init__(self, c, gamma, rcut):
+        self.parameters={}
+        self.parameters['c'] = c
+        self.parameters['gamma'] = gamma
+        self.parameters['rcut'] = rcut
+
+    def value(self, rvec):
+        """
+        Parameters:
+          rvec: (nconf,3) vector
+        Returns:
+          func: c*p(r/rcut)/(1+gamma*p(r/rcut))
+        """
+        r = np.linalg.norm(rvec, axis=-1)
+        y = r/self.parameters['rcut']
+        p = y - y**2 + y**3/3
+        return self.parameters['c'] * p / (1+self.parameters['gamma']*p)
+        
+    def gradient(self, rvec):
+        """
+        Parameters:
+          rvec: (nconf,3) vector
+        Returns:
+          grad: has same dimensions as rvec 
+        """
+        r = np.linalg.norm(rvec, axis=-1, keepdims=True)
```

Comment:
Not necessary for the first version, but just a comment--it's better to combine numpy operations on a single line if possible, since that can avoid the creation of intermediate variables.